stages/authenticator_email: Email Authenticator Stage Documentation (#12853)
goauthentik/authentik: website/docs/add-secure-apps/flows-stages/stages/authenticator_email/index.md

---
title: Email authenticator setup stage
---

> This stage configures an email-based authenticator that sends one-time codes to users' email addresses for authentication.

Reviewer: ```suggestion
This stage configures an email-based authenticator that sends a one-time codes to a user's email address for authentication.
```

## Configuration

### Email Settings

> The stage can be configured in two ways:

Reviewer: ```suggestion
The stage can be configured in two ways:
- global settings
or
- stage-specific settings

```

> #### Global Settings

Reviewer: ```suggestion
#### Global settings
```

> Enable the _Use global settings_ option to use authentik's global email configuration. This is recommended if you already have email configured in authentik.

Reviewer: ```suggestion
Enable the **Use global settings** option to use authentik's global email configuration. This is recommended if you already have email configured in authentik.
```

Author: "Use global settings" is a toggle in the UI, a boolean variable

> #### Stage-specific Settings

Reviewer: ```suggestion
#### Stage-specific settings
```

---

> title: Email authenticator setup stage

Reviewer: ```suggestion
title: Email Authenticator Setup stage
```

Reviewer: I'll add this new decision to our Docs Style Guide, but I think we should capitalize the names of exact stages.

---

> This stage configures an email-based authenticator that sends a one-time codes to a user's email address for authentication.

Reviewer: ```suggestion
This stage configures an email-based authenticator that sends a one-time code to a user's email address for authentication.
```

Reviewer: I missed that first time, sorry.

## Configuration

> ### Email Settings

Reviewer: ```suggestion
### Email settings
```

The stage can be configured in two ways:
- global settings
or
- stage-specific settings

#### Global settings

Enable the **Use global settings** option to use authentik's global email configuration. This is recommended if you already have email configured in authentik.

#### Stage-specific settings

> If you need different email settings for this stage, disable _Use global settings_ and configure the following:

Reviewer: ```suggestion
If you need different email settings for this stage, disable **Use global settings** and configure the following settings:
```

- **Host**: SMTP server hostname (default: localhost)
- **Port**: SMTP server port (default: 25)
- **Username**: SMTP authentication username (optional)
- **Password**: SMTP authentication password (optional)
- **Use TLS**: Enable TLS encryption
- **Use SSL**: Enable SSL encryption
- **Timeout**: Connection timeout in seconds (default: 10)
- **From Address**: Email address messages will be sent from (default: system@authentik.local)

### Message Configuration

> - **Token Expiry**: Time in minutes the sent token is valid (default: 30)

Reviewer: ```suggestion
- **Token Expiry**: Time in minutes that the sent token is valid (default: 30)
```

- **Subject**: Email subject line (default: "authentik Sign-in code")

> ## Flow Integration

Reviewer: ```suggestion
## Flow integration
```

> To use the Email authenticator in a flow:

Reviewer: ```suggestion
To use the Email authenticator setup stage in a flow:
```

> 1. Create a new flow or edit an existing one

Reviewer: ```suggestion
1. [Create](../../flow/#create-a-custom-flow) a new flow or edit an existing one.
```

Reviewer: Sorry, I have low-confidence in that URL I entered above. Please test.

> 2. Add the Email Authenticator Setup Stage

Reviewer: ```suggestion
2. On the flow's **Stage Bindings** tab, add the Email Authenticator Setup stage.
```

3. Configure the stage settings as described above

> When a user goes through this flow, they will:

Reviewer: I think this section, describing what happens after you add this stage to a flow, should go up top, in the "conceptual" section. Then have the procedural section (how to create the stage and add to a flow).
